### ISSUE TITLE
[RFC]: add `@stdlib/array/base/mskreject-map`

### ISSUE THREAD

**Issue by kgryte** [stdlib-js/stdlib#1323]
### Description

This RFC proposes adding the package `@stdlib/array/base/mskreject-map`.

The package should be structured similarly to [`@stdlib/array/base/mskreject`](https://github.com/stdlib-js/stdlib/tree/64931c5fdb4abc3a339b619658d3b5acc089e556/lib/node_modules/%40stdlib/array/base/mskreject).

The package should expose an API having the same signature as `mskreject`, with the addition of `clbk` and `thisArg` arguments.

```javascript
function mskrejectMap( x, mask, clbk[, thisArg ] ) {...}
```

The callback function should map each selected value specified by `mask` to the output value.

In short, this package combines `mskreject` and `map` into a single API call.

Package: `@stdlib/array/base/mskreject-map`
Alias: `mskrejectMap`

### Related Issues

Related issues #1321, #1322 .

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

**Comment by itsspriyansh** [comment]
@kgryte 
I have understood the issue and would like to work on it. Please assign this to me.

**Comment by Planeshifter** [comment]
@itsspriyansh Just assigned you; looking forward to your contribution!

**Comment by Ccivam** [comment]
@kgryte Maybe I can work on this but can you give an example.I didnt understand at all.It will be really helpful.

**Comment by bad-in-coding** [comment]
@kgryte I have understood the issue. Can you assign it to me.

**Comment by adityacodes30** [comment]
@Planeshifter i've looked into this extensively - i would like to implement this, please assign

**Comment by bad-in-coding** [comment]
Hi @kgryte , I worked on this issue. Can you please review it and let me know the changes required. Thanks!
